### DISCUSSION TITLE
Copilot activation failed - VScode

### DISCUSSION THREAD

**Discussion by UmrenTV** [community/community#14854] [Copilot Conversations]
I have bought new laptop and I am trying to re-install all. I installed VScode, and Copilot, it asked me for sign-in with my browser, which I did.. and after which I got crossed Copilot icon in the corner and "Copilot Activation Failed" when you hover over it.

The output is this:
[INFO] [auth] [2022-04-14T01:33:52.430Z] Invalid copilot token: invalid expires_at:1649871265 now:1649900032
[ERROR] [default] [2022-04-14T01:33:52.434Z] GitHub Copilot could not connect to server. Extension activation failed: "Failed to get copilot token"

I have tried, uninstall, re-install. Restarted Code.
Restarted PC as well.
I siged out from Github on Code, and re-installed again. Re-authorized again, still same.

I don't know how to fix it. :(

Any help would be appreciated. Thanks

**Comment by davecheney** [comment]
Your clock is set incorrectly; most likely your location which is affecting the timezone. Once you correct your computers clock, copilot will work.

**Comment by Zhaoyang-Ma** [comment]
I have the same trouble with you, and my clock is set correctly

**Reply by davecheney** [reply]
Please double check that you are running the latest version of the extension. https://github.com/github/feedback/discussions/16815

**Reply by Zhaoyang-Ma** [reply]
Thank for your reply. I didn't do anything with copilot, but it's working now. From yesterday afternoon to this morning, it was not working, and the reason for the error was "GitHub Copilot could not connect to server. Extension activation failed: "self signed certificate""

**Reply by davecheney** [reply]
It looks like someone is interfering with your ssl traffic

**Reply by Zhaoyang-Ma** [reply]
My copilot failed again! What is wrong with it?
![image](https://user-images.githubusercontent.com/32972237/171316338-f99bc5fc-7978-4485-a16c-7d9bd2b4d9c0.png)

**Reply by davecheney** [reply]
Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?

**Reply by Zhaoyang-Ma** [reply]
I was able to use it before, and this is the second time this has happened. The last time this happened, I didn't do anything and it automatically cleared up after a day.

**Reply by davecheney** [reply]
Copilot does not use a self signed certificate, so the presence of one is an indication that someone is intercepting your SSL traffic.
